log4j-api/src/main/java/org/apache/logging/log4j/util/PropertiesUtil.java

```diff
@@ -540,18 +552,18 @@ private synchronized void reload() {
                 final List<CharSequence> tokens = PropertySource.Util.tokenize(key);
                 final boolean hasTokens = !tokens.isEmpty();
                 sources.forEach(source -> {
```

Comment:
Why do we only guard against the particular failure cases reported rather than the entire block? That is, I was expecting:
```
keys.stream().filter(Objects.nonNull).forEach(key -> {
    try {
       // Old logic goes here verbatim
    } catch (final Exception error) {
      // Report the failure
    }
});
for (final PropertySource source : sources) {
    try {
       // Old logic goes here verbatim
    } catch (final Exception error) {
      // Report the failure
    }
}
```

Reply:
Reporting failures will actually generate bug reports. It is very likely that Log4j will try to load the Spring PropertySource before the Spring environment exists. As I recall that will cause an exception during startup of every Spring Boot application, which obviously is unacceptable.

Reply:
If a single property source fails, we don't want the entire lookup to fail.

Anyway I am not a big fan of this _preemptive_ caching strategy adopted by `PropertiesUtil`. In practice the caching process is quite slow, since the values of properties such a `user.home` or all environment variables are cached.
On the other hand users rarely use Log4j configuration properties, so most of the dozens of calls to `PropertiesUtil` result in cache misses.